### PR TITLE
Provide a way to search nodes that have violations

### LINKED ISSUE #1020: Violation listing page
Either bring back the violation listing page, or, provide a default saved search to see the violations

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Add additional search options for the node page to search for nodes with violations
## Impacts

The Admin Node search page
## Issue

Fixed #1020
## NEWS file entries

Enhancements
++++++++++++
- Added the ability to search nodes that have violations
## Delete branch after merge

NO

### CODE REVIEW

File: html/pfappserver/root/admin/nodes.tt [hunk 1]
```diff
@@ -77,6 +78,23 @@ form {
               <li><a href="[% c.pf_hash_for(c.controller('Node').action_for('create')) %]">[% l('Create') %]</a></li>
               [%- END -%]
               [%- IF can_access("NODES_READ") -%]
+              <li class="nav-header">[% l('Standard Searches') %]</li>
+              <li>
+                <a href="#" data-toggle="pf-search-form" data-target="advanced">Open Violations </a>
```
Comment (cgx) [inline html/pfappserver/root/admin/nodes.tt]: Localize the value (and remove the ending space).

File: html/pfappserver/root/admin/nodes.tt [hunk 2]
```diff
@@ -77,6 +78,23 @@ form {
...
+                <form>
+                    <input type="hidden" name="searches.0.name" value="violation_status"/>
+                    <input type="hidden" name="searches.0.op" value="is"/>
+                    <input type="hidden" name="searches.0.value" value="open"/>
+                </form>
+              </li>
+              <li>
+                <a href="#" data-toggle="pf-search-form" data-target="advanced">Closed Violations </a>
```
Comment (cgx) [inline html/pfappserver/root/admin/nodes.tt]: Localize the value (and remove the ending space).

**Comment by julsemaan** [comment]
How hard would it be to add the violation status to the node columns ?

**Comment by fdurand** [comment]
Violation name in a column should be cool

**Comment by louismunro** [comment]
See if you can add the feature requested by @fdurand please. 
Let me know if it's a problem.

**Comment by jrouzierinverse** [comment]
No since there can be multiple violations open it does not make sense.

**Comment by extrafu** [comment]
We could just have a visual indicator in the violation "column" if there's at least one open. If the op wants to know what are they, he/she asks for the detailed view and immediately becomes zammit-newhoodie-happy.

**Comment by jrouzierinverse** [comment]
violation count?

**Comment by jrouzierinverse** [comment]
I will use the same visual indicator like the users page

image

**Comment by extrafu** [comment]
A count could be slow to execute. Think about Aidan showing 1,000 nodes per page - doing a select count for each of them would be painfully slow.

**Comment by fdurand** [comment]
Not violation count but a way to have the name of the violation in a column.
You can see that i have more than one time the same mac address but i don't know what is the violation.
capture du 2016-01-04 16 02 48

**Comment by jrouzierinverse** [comment]
last open violation?

**Comment by fdurand** [comment]
mysql> select \* from violation where mac="00:18:f8:70:bc:37";
+----+-------------------+---------+---------------------+---------------------+--------+------------+-------+
| id | mac               | vid     | start_date          | release_date        | status | ticket_ref | notes |
+----+-------------------+---------+---------------------+---------------------+--------+------------+-------+
| 66 | 00:18:f8:70:bc:37 | 1100010 | 2015-12-22 16:54:22 | 2015-12-22 16:54:25 | closed |            |       |
| 68 | 00:18:f8:70:bc:37 | 1100010 | 2015-12-22 23:03:13 | 2015-12-22 23:03:14 | closed |            |       |
| 70 | 00:18:f8:70:bc:37 | 1100010 | 2015-12-23 05:24:35 | 2015-12-23 05:24:36 | closed |            |       |
| 72 | 00:18:f8:70:bc:37 | 1100010 | 2015-12-23 11:31:08 | 2015-12-23 11:31:09 | closed |            |       |
| 74 | 00:18:f8:70:bc:37 | 1100010 | 2015-12-23 14:53:05 | 2015-12-23 14:53:07 | closed |            |       |
| 76 | 00:18:f8:70:bc:37 | 1100010 | 2015-12-23 17:13:29 | 2015-12-23 17:13:29 | closed |            |       |
| 78 | 00:18:f8:70:bc:37 | 1100010 | 2015-12-23 20:55:05 | 2015-12-23 20:55:05 | closed |            |       |
| 80 | 00:18:f8:70:bc:37 | 1100010 | 2015-12-24 19:24:09 | 2015-12-24 19:24:09 | closed |            |       |
| 82 | 00:18:f8:70:bc:37 | 1100010 | 2015-12-28 06:44:08 | 2015-12-28 06:44:08 | closed |            |       |
| 83 | 00:18:f8:70:bc:37 | 1100010 | 2015-12-28 10:32:49 | 2015-12-28 10:32:49 | closed |            |       |
+----+-------------------+---------+---------------------+---------------------+--------+------------+-------+
10 rows in set (0.00 sec)

mysql> select \* from class where vid=1100010;
+---------+-------------+-------------+-------------+--------------+--------+--------+----------+-----------+----------------+--------------+----------------+---------+-----------+-----------------+----------+------------------+
| vid     | description | auto_enable | max_enables | grace_period | window | vclose | priority | template  | max_enable_url | redirect_url | button_text    | enabled | vlan      | target_category | delay_by | external_command |
+---------+-------------+-------------+-------------+--------------+--------+--------+----------+-----------+----------------+--------------+----------------+---------+-----------+-----------------+----------+------------------+
| 1100010 | Rogue DHCP  | Y           |           3 |         7200 | 0      |   NULL |        4 | roguedhcp | NULL           | NULL         | Enable Network | Y       | isolation | NULL            |        0 | NULL             |
+---------+-------------+-------------+-------------+--------------+--------+--------+----------+-----------+----------------+--------------+----------------+---------+-----------+-----------------+----------+------------------+
1 row in set (0.00 sec)

So the column should contain Rogue DHCP in all case but should be different based on the vid.

**Comment by jrouzierinverse** [comment]
I think we should talk tomorrow when I am in the office.

**Pull request merged**